### PR TITLE
feat: add folder deleted trigger

### PR DESCRIPTION
**Pull request by ZEROM22**

### CODE REVIEW

File: app/files.js [hunk 1]
```diff
@@ -261,6 +273,43 @@ const folderCreated = {
...
+    async perform(z, bundle) {
+      const client = new Client(bundle.authData.baseUrl, z.request)
+      const files = new FilesService(client)
+      const filters = {
+        sortBy: "DateAndTime",
+        sortOrder: "descending",
+        filterType: "FoldersOnly"
+      }
+      const trash = await files.listTrash(filters)
+      if (bundle.inputData.id) return trash.folders.filter(item => item.originRoomId === bundle.inputData.id)
```
Comment (vanyauhalin) [inline app/files.js:306]: Please avoid using one-liners as they can be difficult to read. To help with this, I suggest adding an eslint rule to enforce this guideline.

Comment (vanyauhalin) [inline app/files.js:306]: Please avoid omitting the curly braces for function parameters. To help with this, I suggest adding an eslint rule to enforce this guideline.

File: app/files.test.js [hunk 2]
```diff
@@ -207,4 +208,20 @@ Files("triggers when a room is archived", async (context) => {
...
+    id: context.inputData.folderId
+  }*/
+  const bundle = {
+    authData: context.authData
+  }
+  const folders = await tester(perform, bundle)
+  const folder = folders[0]
+  // TODO: Without the folder delete action, the folder delete trigger test may fail because the trash may be empty.
+  if (folders.length) not.equal(folder.id, 0)
+  else equal(true, true)
```
Comment (vanyauhalin) [inline app/files.test.js:224]: Please avoid using one-liners as they can be difficult to read. To help with this, I suggest adding an eslint rule to enforce this guideline.

File: app/files.js (hunk 1 above)
Comment (vanyauhalin) [inline app/files.js:306]: I'm not sure why the ID field isn't required. Could you please explain it to me?

**Review by vanyauhalin — CHANGES_REQUESTED** [review]

File: app/files.js (hunk 1 above)
Comment (ZEROM22) [reply app/files.js]: > Please avoid using one-liners as they can be difficult to read. To help with this, I suggest adding an eslint rule to enforce this guideline.

Corrected. The eslint rule will be added in a separate PR.

Comment (ZEROM22) [reply app/files.js]: > Please avoid omitting the curly braces for function parameters. To help with this, I suggest adding an eslint rule to enforce this guideline.

Corrected. The eslint rule will be added in a separate PR.

File: app/files.test.js (hunk 2 above)
Comment (ZEROM22) [reply app/files.test.js]: > Please avoid using one-liners as they can be difficult to read. To help with this, I suggest adding an eslint rule to enforce this guideline.

Corrected. The eslint rule will be added in a separate PR.

File: app/files.js (hunk 1 above)
Comment (ZEROM22) [reply app/files.js]: > I'm not sure why the ID field isn't required. Could you please explain it to me?

ID is an additional field to track the removal of a folder from a particular room. Without it, the trigger will be triggered when any folder is deleted.

Comment (vanyauhalin) [reply app/files.js]: It might be better to separate into two different triggers. What do you think?

**Review by vanyauhalin — APPROVED** [review]

**Pull request merged**